### PR TITLE
DEV #18375: Security : Some plugin settings need encryption

### PR DESCRIPTION
**Pull request by gabrieljenik**
New tests for settings set at survey level

### CODE REVIEW

**Review by Shnoulle — APPROVED** [review]

File: tests/functional/backend/SettingsPluginTest.php [hunk 1]
```diff
@@ -75,43 +77,112 @@ public function testGetAndSetSetting()
...
+            $this->assertEquals($settingValue, json_decode(json_encode($value), true));
         }
     }
 
+    /**
+     * Testing for encrypted settings at a survey level
+     */
+    public function testGetAndSetSurveySettingEncripted()
+    {
+        self::$plugin->setEncriptedSettings(array_keys(self::$settings));
```
Comment (olleharstedt) [inline tests/functional/backend/SettingsPluginTest.php:154]: setEncriptedSettings

Spelling error

**Review by olleharstedt — CHANGES_REQUESTED** [review]

File: tests/functional/backend/SettingsPluginTest.php (hunk 1 above)
Comment (olleharstedt) [reply tests/functional/backend/SettingsPluginTest.php:154]: How can the test run correctly when the function is misspelled?

Comment (gabrieljenik) [reply tests/functional/backend/SettingsPluginTest.php:154]: It is misspelled both: on th definition and invocation :)

Comment (olleharstedt) [reply tests/functional/backend/SettingsPluginTest.php:154]: OK can we fix that? :joy:

Comment (gabrieljenik) [reply tests/functional/backend/SettingsPluginTest.php]: Done

Comment (Shnoulle) [reply tests/functional/backend/SettingsPluginTest.php]: It was fixed in https://github.com/LimeSurvey/LimeSurvey/pull/2753 too.

**Pull request merged**